pytest_bdd/utils.py

```diff
@@ -84,3 +84,29 @@ def get_request_fixture_names(request):
     Compatibility with pytest 3.0.
     """
     return request._pyfuncitem._fixtureinfo.names_closure
+
+
+def get_parametrize_markers_args(node):
+    """In pytest 3.6 new API to access markers has been introduced and it deprecated
+    MarkInfo objects.
+
+    This function uses that API if it is available otherwise it uses MarkInfo objects.
+    """
+    mark_name = 'parametrize'
+    try:
+        return get_markers_args_using_iter_markers(node, mark_name)
+    except AttributeError:
+        return get_markers_args_using_get_marker(node, mark_name)
+
+
+def get_markers_args_using_iter_markers(node, mark_name):
+    """Recommended on pytest>=3.6"""
+    args = []
```

Comment:
```suggestion
    return [arg for mark in node.iter_markers(mark_name) for arg in mark.args]
```

Reply:
I usually don't use list comprehension for cases with two loops as in my opinion it harms readability. But if you really want I am happy to do the change.

We would also have to convert it to tuple here.

Reply:
Is it really necessary to convert it to tuple? As you wish, it is just flavour.

Reply:
`get_markers_args_using_get_marker` returns `tuple`. We don't have to convert it to the `tuple` but then sometimes we would return `list` and sometimes `tuple` by calling `get_parametrize_markers_args`.

But you are right. It is not necessary (tests are still passing if we return it as a `list`).

Reply:
I see. Then no change is necessary.